### PR TITLE
rpm: add missing tmpfiles.d configuration

### PR DESCRIPTION
**Pull request by kenhys**

### CODE REVIEW

File: td-agent/templates/usr/lib/tmpfiles.d/td-agent.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+d /tmp/td-agent 0755 td-agent td-agent - -
```
Comment (cosmo0920) [inline td-agent/templates/usr/lib/tmpfiles.d/td-agent.conf:1]: Should we use `@PACKAGE` or `<%= project_name %>` for generating tempfiles configuration?

Comment (kenhys) [reply td-agent/templates/usr/lib/tmpfiles.d/td-agent.conf:1]: Ah, It should be, I'll fix it.

Comment (kenhys) [reply td-agent/templates/usr/lib/tmpfiles.d/td-agent.conf]: Fixed.

File: td-agent/Rakefile [hunk 2]
```diff
@@ -326,6 +326,7 @@ class BuildTask
           "etc/td-agent/td-agent.conf",
           "etc/td-agent/td-agent.conf.tmpl",
           "etc/logrotate.d/td-agent",
+          "usr/lib/tmpfiles.d/td-agent.conf",
```
Comment (cosmo0920) [inline td-agent/Rakefile:329]: CentOS 6 is not systemd based. This should be needed for CentOS 7 and CentOS 8. Not for CentOS 6.

Comment (cosmo0920) [reply td-agent/Rakefile:329]: NOTE: systemd-tmpfiles is the successor systemd module of tmpwatch.
https://www.itmedia.co.jp/help/tips/linux/l0609.html

File: td-agent/yum/td-agent.spec.in [hunk 3]
```diff
@@ -134,6 +134,7 @@ fi
 %{_sysconfdir}/@PACKAGE@/@PACKAGE@.conf.tmpl
 %if %{use_systemd}
 %{_unitdir}/@PACKAGE@.service
+%{_usr}/lib/tmpfiles.d/@PACKAGE@.conf
```
Comment (cosmo0920) [inline td-agent/yum/td-agent.spec.in:137]: We can use %_tmpfilesdir  for %{_usr}/lib/tmpfiles.d/.

File: td-agent/Rakefile (hunk 2 above)
Comment (cosmo0920) [reply td-agent/Rakefile:340]: Hmm..., tmpwatch is a bit messy for modifying installation phase....

**Comment by kenhys** [comment]
CentOS 6 fresh install:

```
$ sudo rpm -ivh td-agent-3.7.1-1.el6.x86_64.rpm 
Preparing...                ########################################### [100%]
   1:td-agent               ########################################### [100%]
[vagrant@localhost ~]$ ls /tmp/td-agent
[vagrant@localhost ~]$ ls -la /tmp/td-agent
total 8
drwxr-xr-x. 2 td-agent td-agent 4096 Jun 10 02:04 .
drwxrwxrwt. 4 root     root     4096 Jun 10 02:04 ..
prw-r--r--. 1 root     root        0 Jun 10 02:04 .keep

```

CentOS 6 upgrade:

```
$ rpm -q td-agent
td-agent-3.7.0-0.el6.x86_64
[vagrant@localhost ~]$ sudo rpm -Uvh td-agent-3.7.1-1.el6.x86_64.rpm 
Preparing...                ########################################### [100%]
   1:td-agent               ########################################### [100%]
[vagrant@localhost ~]$ ls -la /tmp/td-agent
total 8
drwxr-xr-x. 2 td-agent td-agent 4096 Jun 10 02:06 .
drwxrwxrwt. 4 root     root     4096 Jun 10 02:05 ..
prw-r--r--. 1 root     root        0 Jun 10 02:06 .keep
```

**Comment by kenhys** [comment]
CentOS 7 fresh install:

```
$ sudo rpm -ivh td-agent-3.7.1-1.el7.x86_64.rpm 
Preparing...                          ################################# [100%]
Updating / installing...
   1:td-agent-3.7.1-1.el7             ################################# [100%]
$ ls -la /tmp/td-agent/
total 0
drwxr-xr-x. 2 td-agent td-agent   6 Jun 10 01:50 .
drwxrwxrwt. 9 root     root     188 Jun 10 02:09 ..
[vagrant@localhost ~]$ rpm --eval "%{_tmpfilesdir}"
/usr/lib/tmpfiles.d
[vagrant@localhost ~]$ ls /usr/lib/tmpfiles.d/td-agent.conf 
/usr/lib/tmpfiles.d/td-agent.conf
```

CentOS 7 upgrade:
```
$ sudo rpm -Uvh td-agent-3.7.1-1.el7.x86_64.rpm 
Preparing...                          ################################# [100%]
Updating / installing...
   1:td-agent-3.7.1-1.el7             ################################# [ 50%]
Warning: td-agent.service changed on disk. Run 'systemctl daemon-reload' to reload units.
Cleaning up / removing...
   2:td-agent-3.7.0-0.el7             ################################# [100%]
$ ls -la /usr/lib/tmpfiles.d/td-agent.conf 
-rw-r--r--. 1 root root 79 Jun 10 01:47 /usr/lib/tmpfiles.d/td-agent.conf
```

CentOS 7 remove:

```
$ sudo yum remove td-agent
Failed to set locale, defaulting to C
Loaded plugins: fastestmirror
Resolving Dependencies
--> Running transaction check
---> Package td-agent.x86_64 0:3.7.1-1.el7 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================
 Package                              Arch                               Version                                 Repository                             Size
=============================================================================================================================================================
Removing:
 td-agent                             x86_64                             3.7.1-1.el7                             installed                              48 M

Transaction Summary
=============================================================================================================================================================
Remove  1 Package

Installed size: 48 M
Is this ok [y/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
Warning: RPMDB altered outside of yum.
Warning: td-agent.service changed on disk. Run 'systemctl daemon-reload' to reload units.
  Erasing    : td-agent-3.7.1-1.el7.x86_64                                                                                                               1/1 
  Verifying  : td-agent-3.7.1-1.el7.x86_64                                                                                                               1/1 

Removed:
  td-agent.x86_64 0:3.7.1-1.el7                                                                                                                              

Complete!
```

**Comment by cosmo0920** [comment]
yeah, I've also confirmed that the result: 
https://github.com/clear-code/td-agent-builder/pull/88#issuecomment-641676036
https://github.com/clear-code/td-agent-builder/pull/88#issuecomment-641677669

**Comment by cosmo0920** [comment]
BTW, debian's package script also creates `/tmp/<=% project_name %>`.

https://github.com/clear-code/td-agent-builder/blob/master/td-agent/templates/package-scripts/td-agent/deb/postinst#L62-L64
Should we add tempfiles.d configuration in debian package?
WDYT, @kenhys?

**Comment by kenhys** [comment]
Oh, It should be fixed too.

**Pull request merged**